src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -0,0 +1,498 @@
+import { AfterViewInit, Component, ElementRef, Input, OnDestroy, ViewChild, ViewEncapsulation, HostListener } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { fromEvent, Subscription } from 'rxjs';
+
+import { ContentType, LinkType, Orientation, TourStep } from './guided-tour.constants';
+import { GuidedTourService } from './guided-tour.service';
+import { AccountService } from 'app/core';
+
+@Component({
+    selector: 'jhi-guided-tour',
+    templateUrl: './guided-tour.component.html',
+    styleUrls: ['./guided-tour.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class GuidedTourComponent implements AfterViewInit, OnDestroy {
+    @Input() public topOfPageAdjustment = 0;
+    @Input() public tourStepWidth = 500;
+    @Input() public minimalTourStepWidth = 400;
+    @ViewChild('tourStep', { static: false }) public tourStep: ElementRef;
+    public highlightPadding = 4;
+    public currentTourStep: TourStep | null;
+    public selectedElementRect: DOMRect | null;
+    public isOrbShowing = false;
+
+    private resizeSubscription: Subscription;
+    private scrollSubscription: Subscription;
+
+    readonly IMAGE = ContentType.IMAGE;
+    readonly TEXT = ContentType.TEXT;
+    readonly VIDEO = ContentType.VIDEO;
+
+    // TODO
+    readonly LINK = LinkType.LINK;
+    readonly BUTTON = LinkType.BUTTON;
+
+    constructor(public sanitizer: DomSanitizer, public guidedTourService: GuidedTourService, public accountService: AccountService) {}
+
+    /* Enable tour navigation with keyboard arrows */
+    @HostListener('document:keydown', ['$event'])
+    handleKeyboardEvent(event: KeyboardEvent) {
+        if (event.code === 'ArrowRight' && this.guidedTourService.currentTourStepDisplay <= this.guidedTourService.currentTourStepCount) {
+            this.guidedTourService.nextStep();
+        }
+        if (event.code === 'ArrowLeft' && this.guidedTourService.currentTourStepDisplay > 1) {
+            this.guidedTourService.backStep();
+        }
+        if (event.code === 'Escape') {
+            this.guidedTourService.skipTour();
+        }
+    }
+
+    /**
+     * Subscribe to guidedTourCurrentStepStream and set tour step if the user has the right permission
+     */
+    public ngAfterViewInit(): void {
+        this.guidedTourService.guidedTourCurrentStepStream.subscribe((step: TourStep) => {
+            this.currentTourStep = step;
+            if (step) {
+                let hasPermission = true;
```

Comment:
Better instead:
`const hasPermission = this.accountService.hasAnyAuthorityDirect(step.permission)`